### PR TITLE
OSAC-434, OSAC-429, OSAC-433, OSAC-435, OSAC-436, OSAC-437 for Crypto…

### PR DESCRIPTION
**Pull request by alcxs**
…graphy package

### CODE REVIEW

File: Activities/Cryptography/UiPath.Cryptography.Activities/DecryptFile.cs [hunk 1]
```diff
@@ -172,6 +181,31 @@ protected override void Execute(CodeActivityContext context)
                 {
                     fileName = outputFileName;
                 }
+                else
+                {
+                    fileName = Path.GetFileNameWithoutExtension(inputFilePath) + "_Decrypted" + Path.GetExtension(inputFilePath);
```
Comment (ovidiuiliescu) [inline Activities/Cryptography/UiPath.Cryptography.Activities/DecryptFile.cs]: `_Decrypted` should be a const.

File: Activities/Cryptography/UiPath.Cryptography.Activities/DecryptFile.cs [hunk 2]
```diff
@@ -172,6 +181,31 @@ protected override void Execute(CodeActivityContext context)
...
+                    if (string.IsNullOrEmpty(Path.GetDirectoryName(inputFilePath)))
+                    {
+                        filePath = fileName;
+                    }
+                    else
+                    {
+                        filePath = Path.GetFullPath(Path.GetDirectoryName(inputFilePath)) + "\\" + fileName;
```
Comment (ovidiuiliescu) [inline Activities/Cryptography/UiPath.Cryptography.Activities/DecryptFile.cs]: Use path composition methods, don't hardcode the separator here as it's different between OS-es.

Extract the file composition code to a helper method, use it throughout the activities.

File: Activities/Cryptography/UiPath.Cryptography.Activities/EncryptText.cs [hunk 3]
```diff
@@ -117,8 +130,15 @@ protected override string Execute(CodeActivityContext context)
                     throw new ArgumentNullException(Resources.Activity_KeyedHashText_Property_KeySecureString_Name);
                 }
 
+                if (keyEncoding == null && string.IsNullOrEmpty(keyEncodingString)) throw new ArgumentNullException(Resources.Encoding);
+
                 if (keyEncoding == null)
-                    throw new ArgumentNullException(Resources.Encoding);
+                {
+                    //for modern and cross projects - use the string code page to get the encoding
```
Comment (ovidiuiliescu) [inline Activities/Cryptography/UiPath.Cryptography.Activities/EncryptText.cs]: This block repeats throughout, please consolidate in a helper for consistency.

File: Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptFileViewModel.cs [hunk 4]
```diff
@@ -164,5 +200,31 @@ private void KeyInputModeChanged_Action()
...
         }
+
+        /// <summary>
+        /// File input Mode has changed. Set controls visibility based on selection
+        /// </summary>
+        private void FileInputModeChanged_Action()
+        {
+            switch (FileInputModeSwitch.Value)
+            {
+                case FileInputMode.File:
```
Comment (ovidiuiliescu) [inline Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptFileViewModel.cs:209]: Suggestion. Do a ResetAll() before the switch, in the switch just set the visible and required flags for those properties that qualify.

**Review by ovidiuiliescu — CHANGES_REQUESTED** [review]

File: Activities/Cryptography/UiPath.Cryptography.Activities/DecryptFile.cs (hunk 1 above)
Comment (alcxs) [reply Activities/Cryptography/UiPath.Cryptography.Activities/DecryptFile.cs]: done

File: Activities/Cryptography/UiPath.Cryptography.Activities/DecryptFile.cs (hunk 2 above)
Comment (alcxs) [reply Activities/Cryptography/UiPath.Cryptography.Activities/DecryptFile.cs]: done

File: Activities/Cryptography/UiPath.Cryptography.Activities/EncryptText.cs (hunk 3 above)
Comment (alcxs) [reply Activities/Cryptography/UiPath.Cryptography.Activities/EncryptText.cs]: done

File: Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptFileViewModel.cs (hunk 4 above)
Comment (alcxs) [reply Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptFileViewModel.cs:209]: done

File: Activities/Cryptography/UiPath.Cryptography.Activities/EncryptFile.cs [hunk 5]
```diff
@@ -175,18 +187,47 @@ protected override void Execute(CodeActivityContext context)
                 {
                     fileName = outputFileName;
                 }
+                else
+                {
+                    fileName = Path.GetFileNameWithoutExtension(inputFilePath) + Encrypted + Path.GetExtension(inputFilePath);
```
Comment (ovidiuiliescu) [inline Activities/Cryptography/UiPath.Cryptography.Activities/EncryptFile.cs]: This block and the one in DecryptFile are functionally idential, just the arguments are different. Please consolidate them in a helper.

Comment (alcxs) [reply Activities/Cryptography/UiPath.Cryptography.Activities/EncryptFile.cs]: fixed

File: Activities/Cryptography/UiPath.Cryptography.Activities/KeyedHashFile.cs [hunk 6]
```diff
@@ -154,6 +165,14 @@ protected override string Execute(CodeActivityContext context)
                     filePath = localFile.LocalPath;
                 }
 
+                if (keyEncoding == null)
```
Comment (radunegulescu2) [inline Activities/Cryptography/UiPath.Cryptography.Activities/KeyedHashFile.cs]: Shouldn't the helper function KeyEncodingOrString be applied here?

Comment (alcxs) [reply Activities/Cryptography/UiPath.Cryptography.Activities/KeyedHashFile.cs]: done, thanks for spot

**Review by ovidiuiliescu — APPROVED** [review]

**Pull request merged**